tiledb/api/src/array/dimension/mod.rs

```diff
@@ -521,216 +270,22 @@ impl From<Builder> for Dimension {
-                let strat_bound =
-                    proptest::string::string_regex("[ -~]*").unwrap().boxed();
-
-                if cell_bound.is_some() {
-                    /*
-                     * This is not tractible unless there is a bound on the string length.
-                     * There isn't one since `StringAscii` is only allowed as a dimension
-                     * type in sparse arrays.
-                     */
-                    return None;
-                }
-
-                let strat = (strat_bound.clone(), strat_bound).prop_map(
-                    |(ascii1, ascii2)| {
-                        let (lb, ub) = if ascii1 < ascii2 {
-                            (ascii1, ascii2)
-                        } else {
-                            (ascii2, ascii1)
-                        };
-                        Range::Var(VarValueRange::from((lb, ub)))
-                    },
-                );
-                Some(strat.boxed())
-            }
-        )
-    }
-}
-
-impl Display for DimensionData {
-    fn fmt(&self, f: &mut Formatter) -> FmtResult {
-        write!(f, "{}", json!(*self))
-    }
-}
-
-impl TryFrom<&Dimension> for DimensionData {
-    type Error = crate::error::Error;
-
-    fn try_from(dim: &Dimension) -> TileDBResult<Self> {
-        let datatype = dim.datatype()?;
-        let constraints = physical_type_go!(datatype, DT, {
-            let domain = dim.domain::<DT>()?;
-            let extent = dim.extent::<DT>()?;
-            if let Some(domain) = domain {
-                DimensionConstraints::from((domain, extent))
-            } else {
-                assert!(extent.is_none());
-                DimensionConstraints::StringAscii
-            }
-        });
-
-        Ok(DimensionData {
-            name: dim.name()?,
-            datatype,
-            constraints,
-            filters: {
-                let fl = FilterListData::try_from(&dim.filters()?)?;
-                if fl.is_empty() {
-                    None
-                } else {
-                    Some(fl)
-                }
-            },
-        })
-    }
-}
-
-impl TryFrom<Dimension> for DimensionData {
-    type Error = crate::error::Error;
-
-    fn try_from(dim: Dimension) -> TileDBResult<Self> {
-        Self::try_from(&dim)
-    }
-}
-
-impl Factory for DimensionData {
-    type Item = Dimension;
-
-    fn create(&self, context: &Context) -> TileDBResult<Self::Item> {
-        let mut b = Builder::new(
-            context,
-            &self.name,
-            self.datatype,
-            self.constraints.clone(),
-        )?;
-
-        if let Some(fl) = self.filters.as_ref() {
-            b = b.filters(fl.create(context)?)?;
-        }
-
-        Ok(b.cell_val_num(self.cell_val_num())?.build())
-    }
-}
-
 #[cfg(feature = "arrow")]
 pub mod arrow;
 
-#[cfg(any(test, feature = "proptest-strategies"))]
-pub mod strategy;
+#[cfg(any(test, feature = "pod"))]
+pub mod pod;
```

Comment:
Still not sure on this pod as feature thing. Also, we should probably standardize that all module definitions (that aren't `tests`) should go at the top of the file. Finding the code/test transition isn't the easiest thing to navigate to just to check if there's anything interesting going on. Don't worry about that as part of this PR. Just another thing I've noticed while reading through.

Reply:
Yeah I think we have some stylization to do
- modules at the top
- `use` declaration un-grouping